src/me/lotabout/codegenerator/config/include/Include.java

```diff
@@ -0,0 +1,133 @@
+package me.lotabout.codegenerator.config.include;
+
+import com.intellij.openapi.util.io.FileUtil;
+import org.apache.commons.lang.builder.EqualsBuilder;
+import org.apache.commons.lang.builder.HashCodeBuilder;
+
+import javax.xml.bind.annotation.*;
+import java.io.IOException;
+import java.util.UUID;
+
+@XmlRootElement(name = "include")
+@XmlAccessorType(XmlAccessType.FIELD)
+public class Include {
+    @XmlAttribute
+    public static final String VERSION = "1.3";
+
+    private UUID id;
+    public String name = "Untitled";
+    public String fileNamePattern = ".*\\.java$";
+    public String content = DEFAULT_TEMPLATE;
+    public String fileEncoding = DEFAULT_ENCODING;
+    public boolean defaultInclude;
+
+    public Include(UUID id) {
+        this.id = id;
+    }
+
+    public Include(String id) {
+        this.id = UUID.fromString(id);
+    }
+
+    public Include() {
+        this(UUID.randomUUID());
+    }
+
+    public void regenerateId() {
+        this.id = UUID.randomUUID();
+    }
+
+    public String getId() {
+        return this.id.toString();
+    }
+
+    public String getName() {
+        return name;
+    }
+
+    public String getContent() {
+        return content;
+    }
+
+    public void setId(UUID id) {
+        this.id = id;
+    }
+
+    public void setName(String name) {
+        this.name = name;
+    }
+
+    public String getFileNamePattern() {
+        return fileNamePattern;
+    }
+
+    public void setFileNamePattern(String fileNamePattern) {
+        this.fileNamePattern = fileNamePattern;
+    }
+
+    public void setContent(String content) {
+        this.content = content;
+    }
+
+    public String getFileEncoding() {
+        return fileEncoding;
+    }
+
+    public void setFileEncoding(String fileEncoding) {
+        this.fileEncoding = fileEncoding;
+    }
+
+    public boolean isDefaultInclude() {
+        return defaultInclude;
+    }
+
+    public void setDefaultInclude(boolean defaultInclude) {
+        this.defaultInclude = defaultInclude;
+    }
+
+    public static final String DEFAULT_ENCODING = "UTF-8";
+
+    private static final String DEFAULT_TEMPLATE;
+
+    static {
+        String default_template;
+        try {
+            default_template = FileUtil.loadTextAndClose(Include.class.getResourceAsStream("/template/default.vm"));
```

Comment:
@ramazangirgin 

1. Please provide a default template for `include` that explains what `include` does, an example, and how it is done.
2. Firstly I thought includes are just preamble, but it seems that they'll be parsed? doesn't quite catch its use cases.
3. Also I found that `#` characters could not be present in include definitions?

Reply:
@lotabout 

1. Actually Include templates are used for defining shared code templates which can be used inside multiple code generation templates. When include template is defined, it could be included in code generator template with the expression #parse("INCLUDE_NAME")
it is similar to velocity template engine parse support https://velocity.apache.org/engine/1.7/user-guide.html#parse
I will add sample include file.
2. Yes, include template are parsed when  #parse("INCLUDE_NAME") expression is used in code generator template or in another include template
3. No # characters can be used also in include definitions. 
For example; it can be used for parsing another include template or defining a macro with macro expression #macro